docs: update throttling doc (#3545)
Azure/aks-engine: docs/topics/azure-api-throttling.md

- 1.17.5
- 1.18.2

We will demonstrate a real-time, in place remediation that updates the Azure cloud provider runtimes to a newer version. This requires that we stop all active Kubernetes components that interface with Azure APIs (e.g., kube-controller-manager or cloud-controller-manager, cluster-autoscaler), update those component specs so that they use a newer, optimized version of the Azure cloud provider, wait ~15-30 minutes, and then restart those components.

## Stop kube-controller-manager (or cloud-controller-manager) component

> The easiest way to stop the `kube-controller-manager` (or `cloud-controller-manager` if you're using the Azure cloud provider controller as a separate component) is to move its pod spec out of the directory where the kubelet process continually processes control plane manifests (`/etc/kubernetes/manifests`). The running kubelet on the control plane VM will detect the missing controller-manager spec, and automatically delete the pod scheduled on that node. This has the effect of removing that particular Azure API client runtime from operation, reducing calls that count against subscription + region request limits. Below we will demonstrate how to do that from a control plane VM.

Reviewer: ```suggestion
The easiest way to stop the `kube-controller-manager` (or `cloud-controller-manager` if you're using the Azure cloud provider controller as a separate component) is to move its pod spec out of the directory where the kubelet process continually processes control plane manifests (`/etc/kubernetes/manifests`). The running kubelet on the control plane VM will detect the missing controller-manager spec, and automatically delete the pod scheduled on that node. This has the effect of removing that particular Azure API client runtime from operation, reducing calls that count against subscription-wide and regional request limits. Below we will demonstrate how to do that from a control plane VM.

```

(Note: if managing a cluster-autoscaler installation outside of AKS Engine addons, for example using a Helm chart, do the equivalent version update steps as demonstrated above to the self-managed cluster-autoscaler spec, if appropriate.)

## Restart kube-controller-manager (or cloud-controller-manager) component

Again, we assume our ~30 minute cool down window has passed, and we're ready to restart the controller-manager control plane component by moving the pod spec back to the original `/etc/kubernetes/manifests/` location:

```
azureuser@k8s-master-31453872-0:~$ for control_plane_vm in $(kubectl get nodes | grep k8s-master | awk '{print $1}'); do ssh $control_plane_vm "sudo mv /opt/azure/kube-controller-manager.yaml /etc/kubernetes/manifests/kube-controller-manager.yaml"; done

Authorized uses only. All activity may be monitored and reported.

Authorized uses only. All activity may be monitored and reported.

Authorized uses only. All activity may be monitored and reported.
```

> ## Verify controller-manager is re-loaded, running the desired, upgraded version

Reviewer: ```suggestion
## Verify controller-manager is re-loaded and running the desired, upgraded version

```

Now, we can verify that v1.15.12 is the running `controller-manager` version:

```

> azureuser@k8s-master-31453872-0:~$ for pod in $(kubectl get pods -n kube-system -l component=kube-controller-manager | awk '{print $1}'); do kubectl logs $pod -n kube-system | grep "v1.15.12"; done

Reviewer: 👍

I0624 18:32:18.102332       1 controllermanager.go:164] Version: v1.15.12
I0624 18:32:45.174767       1 controllermanager.go:164] Version: v1.15.12
I0624 18:33:05.825467       1 controllermanager.go:164] Version: v1.15.12
```

## Reload cluster-autoscaler addon

If we're also using (and have stopped, and/or updated) `cluster-autoscaler`, remember to reload it, and/or verify that it's running the desired version. For example, restore the replica count to its original value:

```
azureuser@k8s-master-31453872-0:~$ kubectl edit deployment cluster-autoscaler -n kube-system
...
```

> Now you can validate that it's running, and using the desired version. E.g.:

Reviewer: ```suggestion
Now you can validate that it's running with the desired version. For example:

```

```
azureuser@k8s-master-31453872-0:~$ for pod in $(kubectl get pods -n kube-system -l app=cluster-autoscaler | awk '{print $1}'); do kubectl logs $pod -n kube-system | grep "1.15"; done
I0624 18:32:40.458155       1 main.go:354] Cluster Autoscaler 1.15.6
```

> Again, there are a variety of `cluster-autoscaler` implementations that our cluster may be using, either a user-configured AKS Engine-provided cluster-autoscaler, or a Helm chart, as two examples.

Reviewer: ```suggestion
Again, there are a variety of `cluster-autoscaler` implementations that our cluster may be using. Two examples are the user-configurable cluster-autoscaler addon provided by AKS Engine and a Helm chart.
```
